web/src/app/schedules/ScheduleTZFilter.tsx

```diff
@@ -13,25 +13,46 @@ const tzQuery = gql`
     }
   }
 `
+interface ScheduleTZFilterProps {
+  scheduleID: string
+  label?: string | ((tz: string) => string)
+  render?: (children: JSX.Element) => JSX.Element
+}
 
-export function ScheduleTZFilter(props) {
+export function ScheduleTZFilter(
+  props: ScheduleTZFilterProps,
+): JSX.Element | null {
   const [zone, setZone] = useURLParam('tz', 'local')
   const { data, loading, error } = useQuery(tzQuery, {
     pollInterval: 0,
```

Comment:
I think if we were to remove this line we would get the logic desired to fix the current bug + move forward with this PR

Reply:
Needs a slight bit more ^, but it's working so far with using an Effect as well to watch for changes:

```suggestion
  const [filterTZ, setFilterTZ] = useURLParam('tz', 'local')
  const resetTZ = useResetURLParams('tz')
  const { data, loading, error } = useQuery(tzQuery, {
    variables: { id: props.scheduleID },
  })
  const scheduleTZ = data?.schedule?.timeZone
  const localTZ = DateTime.local().zone.name

  useEffect(() => {
    if (filterTZ === localTZ || localTZ === scheduleTZ) {
      resetTZ()
    }
  }, [scheduleTZ])

  if (localTZ === scheduleTZ) return null
```